addons/adyen_platforms/models/adyen_bank_account.py

```diff
@@ -0,0 +1,152 @@
+        country_id = self.env['res.country'].browse(values.get('country_id')) if values.get('country_id') else self.country_id
+        currency_id = self.env['res.currency'].browse(values.get('currency_id')) if values.get('currency_id') else self.currency_id
+        owner_country_id = self.env['res.country'].browse(values.get('owner_country_id')) if values.get('owner_country_id') else self.owner_country_id
+        owner_state_id = self.env['res.country.state'].browse(values.get('owner_state_id')) if values.get('owner_state_id') else self.owner_state_id
+        return {
+            'accountHolderCode': adyen_account_id.account_holder_code,
+            'accountHolderDetails': {
+                'bankAccountDetails': [{
+                    'accountNumber': values.get('account_number') or self.account_number or None,
+                    'accountType': values.get('account_type') or self.account_type or None,
+                    'bankAccountReference': values.get('bank_account_reference') or self.bank_account_reference,
+                    'bankAccountUUID': values.get('bank_account_uuid') or self.bank_account_uuid or None,
+                    'bankCity': values.get('bank_city') or self.bank_city or None,
+                    'bankCode': values.get('bank_code') or self.bank_code or None,
+                    'bankName': values.get('bank_name') or self.bank_name or None,
+                    'branchCode': values.get('branch_code') or self.branch_code or None,
+                    'countryCode': country_id.code,
+                    'currencyCode': currency_id.name,
+                    'iban': values.get('iban') or self.iban or None,
+                    'ownerCity': values.get('owner_city') or self.owner_city or None,
+                    'ownerCountryCode': owner_country_id.code or None,
+                    'ownerHouseNumberOrName': values.get('owner_house_number_or_name') or self.owner_house_number_or_name or None,
+                    'ownerName': values.get('owner_name') or self.owner_name,
+                    'ownerPostalCode': values.get('owner_zip') or self.owner_zip or None,
+                    'ownerState': owner_state_id.code or None,
+                    'ownerStreet': values.get('owner_street') or self.owner_street or None,
+                }],
+            }
+        }
+
+    def _upload_bank_statement(self, content, filename):
+        content_encoded = content.encode('utf8')
+        mimetype = guess_mimetype(base64.b64decode(content_encoded))
+        file_size = len(content_encoded)
+        if mimetype not in ['image/jpeg', 'image/png', 'application/pdf']:
+            raise ValidationError(_('Allowed file formats for bank statements are jpeg, jpg, pdf or png'))
+        if file_size >> 20 > 10 or (file_size >> 10 < 10 and mimetype != 'application/pdf'):
+            raise ValidationError(_('Bank statements must be greater than 10kB (except for PDFs) and smaller than 10MB'))
```

Comment:
Greater than 10KB? Either split the error in two cases, either display the size in the error message.